scripts/wmata_data/distcalc.py

```diff
@@ -0,0 +1,149 @@
+def findBusStations(project, radiusinmeters, distCsvWriter):
+    lat = project['Proj_lat']
+    lon = project['Proj_lon']
+
+    params = {'Lat': lat,
+              'Lon' : lon,
+              'Radius': str(radiusinmeters)}
+    response = requests.get('https://api.wmata.com/Bus.svc/json/jStops',params=params, headers=WMATA_HEADERS)
+    data = response.json()
+
+    for stop in data['Stops']:
+        walkDist = getWalkingDistance(lat, lon, str(stop['Lat']), str(stop['Lon']))
+
+        if walkDist <= radiusinmeters: #within 0.5 miles walking
+            distCsvWriter.writerow((project['Nlihc_id'],'bus',stop['StopID'],"{0:.2f}".format(getMiles(walkDist))))
+
+
+####### Writes rail station metadata to given CSV writer
+def writeRailInfo(railStations, infoCsvWriter):
+
+    print("Writing RAIL INFO");
+
+    for station in railStations:
+        #delimit list of lines with colon
+        lines = station['LineCode1'] #all stations have at least one line
+
+        #there is probably a more elegant way to write this...
+        if station['LineCode2'] != None:
+            lines = '{}:{}'.format(lines, station['LineCode2'])
+
+        if station['LineCode3'] != None:
+            lines = '{}:{}'.format(lines, station['LineCode3'])
+
+        if station['LineCode4'] != None:
+            lines = '{}:{}'.format(lines, station['LineCode4'])
+
+        infoCsvWriter.writerow((station['Code'],station['Name'],str(station['Lat']),
+                                                        str(station['Lon']),lines))
+####### Writes bus station metadata to given CSV writer
+def writeBusInfo(infoCsvWriter):
+    print("Writing BUS INFO")
+
+    response = requests.get('https://api.wmata.com/Bus.svc/json/jStops', headers=WMATA_HEADERS)
+    data = response.json()
+
+    for stop in data['Stops']:
+
+        lines = ""
+        for route in stop['Routes']:
+            lines = '{}:{}'.format(lines,route)
+        lines = lines[1:] #take off the first :
+
+        infoCsvWriter.writerow((stop['StopID'],stop['Name'],stop['Lat'],stop['Lon'],lines))
+
+
+
+########### MAIN #############
+def main(csvInputFileName,distOutputFileName,infoOutputFileName):
+
+    projectsFile = open(csvInputFileName)
+    distOutputFile = open(distOutputFileName, 'wt')
+    distCsvWriter = csv.writer(distOutputFile)
+
+    infoOutputFile = open(infoOutputFileName, 'wt')
+    infoCsvWriter = csv.writer(infoOutputFile)
+
+    reader = csv.DictReader(projectsFile)
+
+    numrow = 0
+
+    #print headers
+    distCsvWriter.writerow(('Nlihc_id','type','stop_id_or_station_code','dist_in_miles'))
+    infoCsvWriter.writerow(('code_or_id','type','name','lat','lon','lines'))
+
+    railResponse = requests.get("https://api.wmata.com/Rail.svc/json/jStations",headers =WMATA_HEADERS)
+    railStations = railResponse.json()['Stations']
+    writeRailInfo(railStations, infoCsvWriter)
+    writeBusInfo(infoCsvWriter)
```

Comment:
So I refactored a bit keeping in mind the idea that one might want to call a func from the other module.  The main reason I did it differently was because I use the railStations json object later when computing the distances for the projects, so I don't have to make extra API calls.  But I refactored it so it can accomplish both goals of keeping railStations json for later processing & being able to call the func from other modules if desired.